src/Umbraco.Web.UI.Client/src/views/propertyeditors/nestedcontent/nestedcontent.controller.js

```diff
@@ -208,6 +211,39 @@ angular.module("umbraco").controller("Umbraco.PropertyEditors.NestedContent.Prop
             $scope.labels.content_createEmpty = data[1];
         });
 
+
+        
+        var copyAllEntries = function() {
+            
+            syncCurrentNode();
+
+            // list aliases
+            var aliases = $scope.nodes.map((node) => node.contentTypeAlias);
+
+            // remove dublicates
+            aliases = aliases.filter((item, index) => aliases.indexOf(item) === index);
+
+            // Retrive variant name
+            var culture = $routeParams.cculture ? $routeParams.cculture : $routeParams.mculture;
```

Comment:
This isn't a safe way to get the current culture. We should never rely on the $routeParams except at the top level. This won't work when it's in split view for example. To get the current culture requires a component model, for example see the `valPropertyMsg` directive requires `^^umbVariantContent` ... which exposes the current editor's language (i.e. `umbVariantContent.editor.content`). This is the only way to get the current editors culture when in split view since the routeParams are not sufficient for that and should generally be avoided entirely unless at the highest editor level. This does mean that the nested content controller/editor should be adjusted to be a component. Components can now be stored anywhere so a component for nested content can just exist beside this controller/view it doesn't have to go to the weird components folder.

Reply:
I think this argument makes it clear that we should go with the Component Solution. I'm on it!